### PR TITLE
112 include test coverage

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -27,3 +27,57 @@ jobs:
...
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                path: 'README.md',
+                message: 'Update README',
+                content: Buffer.from(lines.join('\n')).toString('base64'),
+                sha: response.data.sha,
+                branch: context.ref
+              });
+            }
+
```
Comment (gregLibert) [inline .github/workflows/node.js.yml:79]: Can you put that in shell script ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x, 18.x]
+        node-version: [18.x]
```
Comment (gregLibert) [inline .github/workflows/node.js.yml:15]: If you change that, you can remove the matrix

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -27,3 +27,25 @@ jobs:
...
+    - id: coverage
+      run: |
+        value=$(npm run test:ci | grep -oE '^│\s*total\s+\│\s*([0-9.]+)%' | grep -oE '[0-9]+(\.[0-9]+)?%' | cut -d'.' -f1)
+        echo "coverage=$value" >> $GITHUB_OUTPUT
+
+    - name: Update README
+      run: |
+        "${GITHUB_WORKSPACE}/.github/coverage.sh"
+
+    - name: Commit changes
```
Comment (gregLibert) [inline .github/workflows/node.js.yml:39]: i have the feeling that this won't work.
I think you have to:
- do the checkout using massabot account like [here](https://github.com/massalabs/massa-as-sdk/blob/main/.github/workflows/powered-by.yml#L16).
- set write permission like [here](https://github.com/massalabs/massa-as-sdk/blob/main/.github/workflows/powered-by.yml#L11).

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -27,3 +27,25 @@ jobs:
...
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: "Generate coverage badge"
+        file_pattern: "README.md"
+        # Commit code using massabot account that can bypass push and MR restriction
+        commit_author: massabot <info@massa.net>
+ 
+
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (gregLibert) [inline .github/workflows/node.js.yml:49]: Why do youneed github token ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (0xSwapFeeder) [reply .github/workflows/node.js.yml]: To push with the github-actionbot from the script

**Review by gregLibert — APPROVED** [review]

**Pull request merged**